src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py

```diff
@@ -0,0 +1,1033 @@
+            _ = self.unet(latents, t, encoder_hidden_states=text_embeddings[1].unsqueeze(0)).sample
+            self.unet.zero_grad()
+
+            # Get max activation value for each subject token
+            max_attention_per_index = self._aggregate_and_get_max_attention_per_token(
+                indices_to_alter=indices_to_alter,
+            )
+
+            loss, losses = self._compute_loss(max_attention_per_index, return_losses=True)
+
+            if loss != 0:
+                latents = self._update_latent(latents, loss, step_size)
+
+            try:
+                low_token = np.argmax([l.item() if type(l) != int else l for l in losses])
+            except Exception as e:
+                logger.info(e)  # catch edge case :)
+                low_token = np.argmax(losses)
+
+            low_word = self.tokenizer.decode(text_input.input_ids[0][indices_to_alter[low_token]])
+            logger.info(f"\t Try {iteration}. {low_word} has a max attention of {max_attention_per_index[low_token]}")
+
+            if iteration >= max_refinement_steps:
+                logger.info(
+                    f"\t Exceeded max number of iterations ({max_refinement_steps})! "
+                    f"Finished with a max attention of {max_attention_per_index[low_token]}"
+                )
+                break
+
+        # Run one more time but don't compute gradients and update the latents.
+        # We just need to compute the new loss - the grad update will occur below
+        latents = latents.clone().detach().requires_grad_(True)
+        _ = self.unet(latents, t, encoder_hidden_states=text_embeddings[1].unsqueeze(0)).sample
+        self.unet.zero_grad()
+
+        # Get max activation value for each subject token
+        max_attention_per_index = self._aggregate_and_get_max_attention_per_token(
+            indices_to_alter=indices_to_alter,
+        )
+        loss, losses = self._compute_loss(max_attention_per_index, return_losses=True)
+        logger.info(f"\t Finished with loss of: {loss}")
+        return loss, latents, max_attention_per_index
+
+    def register_attention_control(self):
+        attn_procs = {}
+        cross_att_count = 0
+        for name in self.unet.attn_processors.keys():
+            if name.startswith("mid_block"):
+                place_in_unet = "mid"
+            elif name.startswith("up_blocks"):
+                place_in_unet = "up"
+            elif name.startswith("down_blocks"):
+                place_in_unet = "down"
+            else:
+                continue
+
+            cross_att_count += 1
+            attn_procs[name] = AttendExciteCrossAttnProcessor(
+                attnstore=self.attention_store, place_in_unet=place_in_unet
+            )
+
+        self.unet.set_attn_processor(attn_procs)
+        self.attention_store.num_att_layers = cross_att_count
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        prompt: Union[str, List[str]],
+        indices_to_alter: List[int],
```

Comment:
(nit) wonder whether it could make more sense to just call it `indices` to keep things short :-)